src/core/data/allocator.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "core/data/buffer.h"
+
+#include <unordered_map>
+
+namespace legate {
+
+class IntraTaskAllocator {
+ public:
+  using ByteBuffer = Buffer<int8_t>;
+
+ public:
+  IntraTaskAllocator() = default;
+  IntraTaskAllocator(Legion::Memory::Kind kind);
+  virtual ~IntraTaskAllocator();
+
+ public:
+  typedef char value_type;
```

Comment:
Let's use `using` instead.

Reply:
Also, is the contract for a custom allocator for Thrust? Otherwise, we shouldn't add this member.

Reply:
Yes, this is what thrust expects.